codex/stores/cachestore.nim

```diff
@@ -109,23 +109,20 @@ method putBlock*(
 
 method delBlock*(
   self: CacheStore,
-  cid: Cid): Future[bool] {.async.} =
-  ## delete a block/s from the block store
+  cid: Cid): Future[?!void] {.async.} =
+  ## Delete a block from the blockstore
   ##
 
   trace "Deleting block from cache", cid
   if cid.isEmpty:
     trace "Empty block, ignoring"
-    return true
+    return success()
 
-  try:
-    let removed = self.cache.del(cid)
-    if removed.isSome:
-      self.currentSize -= removed.get.data.len
-      return true
-    return false
-  except EmptyLruCacheError:
-    return false
+  let removed = self.cache.del(cid)
```

Comment:
This should use `.catch` to convert the exception to a result, otherwise this will propagate the exception up the stack without much compile time checking.

Reply:
The current implementation of cache.del function doesn't raise EmptyLruCacheError: https://github.com/status-im/lrucache.nim/blob/master/src/lrucache.nim#L75

Overall, it seems that, in this implementation, only functions explicitly dealing with LRU entries, such as removeLru, can raise this exception